theurgy.lic

```diff
@@ -445,21 +435,18 @@ class TheurgyActions
   end
 
   def commune_tamsine
-    return false unless holy_water?
-
-    pause 1
-    waitrt?
-    DRCI.get_item_safe(@water_holder, @theurgy_supply_container)
-    DRC.bput("sprinkle #{@water_holder} on #{checkname}", 'You sprinkle yourself',
-         'Sprinkle what?')
-    match = DRC.bput('commune tamsine',
+    unless DRCTH.sprinkle_holy_water?(@theurgy_supply_container, @water_holder, checkname)
+      DRC.message("Holy water sprinkling routine failed, skipping Tamsine commune")
+      return
+    end
+    case DRC.bput('commune tamsine',
                  'completed this commune too recently',
                  'You struggle to commune',
                  'you have attempted a commune too recently in the past',
                  'You feel warmth spread throughout your body')
-    DRCI.put_away_item?(@water_holder, @theurgy_supply_container)
-
-    match == 'You feel warmth spread throughout your body'
+    when /You feel warmth spread throughout your body/
+      return true
```

Comment:
Doesn't this need `false` at the end to be explicit? I'm not super clear on what the return value of a case/when is when no branch was taken...